recipes-bsp/optee-os/optee-os-iot2050_3.16.0.bb

```diff
@@ -22,7 +28,8 @@ OPTEE_NAME = "iot2050"
 OPTEE_PLATFORM = "k3-am65x"
 OPTEE_EXTRA_BUILDARGS = " \
     CFG_ARM64_core=y CFG_TEE_CORE_LOG_LEVEL=2 CFG_USER_TA_TARGETS=ta_arm64 \
-    CFG_CONSOLE_UART=1"
+    CFG_CONSOLE_UART=1 CFG_RPMB_FS=y CFG_RPMB_FS_DEV_ID=1 CFG_CORE_DYN_SHM=y \
```

Comment:
What about CFG_RPMB_WRITE_KEY in which case it should be added?

Reply:
https://github.com/OP-TEE/optee_os/issues/5338
"CFG_RPMB_WRITE_KEY=y is very dangerous to use a on a production platform."

But let's try to pull in @apalos here as we left our private discussion with the question whether it takes a special firmware version (or rather a special op-tee build) to do the pairing with RPMB. Ilias, can you comment on this?

Looks like we want this `WRITE_KEY` switch to be build-time configurable and tell our users to never ship (or even generate) a _signed_ build which has this switch enabled. Pairing could also be done in a safe environment before enabling secure boot, but it does take a special firmware in my understanding.

Reply:
> Pairing could also be done in a safe environment before enabling secure boot, but it does take a special firmware in my understanding.

Then when the device is leaving the safe environment without secure boot enabled, attacker could burn any OPTee to steal the RPMB key.

Reply:
Or rather any software running at S-EL1. Anyway device need regenerate the RPMB key in secure memory with OPTee code, hacker could reverse engineering on the shipped S-EL1 (normally OPTee) code to get the key-generating logic and replay in their own S-EL1 software to dump the RPMB key.

I think most of the users will use the upstream OPTee code directly, which make it more easier, the key regenerating logic is already there in source code format.

Reply:
My point is when the device is leaving the safe environment for a non-safe circumstance, the secure boot must be enabled.

Reply:
> tell our users to never ship (or even generate) a _signed_ build which has this switch enabled.

What could happen if this *signed* build is leaked?

Reply:
That is exactly what we should understand better. It could be "very dangerous", see above.

Reply:
OK, I thought this is the real risk:

```
+OP-TEE can program this key when built with ``CFG_RPMB_WRITE_KEY=y``. One
+must be well warned that this configuration makes OP-TEE to send the RPMB key
+in clear to the non-secure side that relays the RPMB key programming request to
+the eMMC hardware device.
```

From https://github.com/OP-TEE/optee_docs/pull/157

Reply:
@BaochengSu usually this takes place in production once, so although it will happen in clear text it should at least be in a controlled safe environment. 

The main cause of problems here is how OP-TEE generates the key to begin with.  If the platform does not provide it's own key (ideally per hardware), OP-TEE will end up programming the default key.  However the default key is known and should only be used for testing.  

Unless your platform has it's own implementation of `tee_otp_get_hw_unique_key` it's probably using the default OP-TEE key

Reply:
Yes.  I'll go update the op-tee documentation but basically you got two threats. The first is what I mentioned above.  The second one is related to how OPP-TEE accesses the eMMC to begin with.

OP-TEE uses a supplicant to communicate with the non secure world eMMC driver (since the hardware resides in non-secure world to begin with). If someone replaces the supplicant with a rogue application always replying "rpmb key not programmed", OP-TEE will then send the key in clear text,  asking for the supplicant to program it.